sdcm/utils/benchmarks.py

```diff
@@ -0,0 +1,220 @@
+    """
+
+    def __init__(self):
+        self._nodes: list["BaseNode"] = []
+        self._benchmark_runners: list[ScyllaNodeBenchmarkRunner] = []
+        self._es = ES()
+
+    def add_node(self, new_node: "BaseNode"):
+        if new_node.distro.is_debian_like:
+            self._benchmark_runners.append(ScyllaNodeBenchmarkRunner(new_node))
+        else:
+            LOGGER.debug("Skipped installing benchmarking tools on a non-debian-like distro.")
+
+    def add_nodes(self, nodes: list["BaseNode"]):
+        for node in nodes:
+            self.add_node(node)
+
+    def install_benchmark_tools(self):
+        try:
+            parallel = ParallelObject(self._benchmark_runners, timeout=300)
+            parallel.run(lambda x: x.install_benchmark_tools(), ignore_exceptions=True)
+        except TimeoutError as exc:
+            LOGGER.warning("Ran into TimeoutError while installing benchmark tools: Exception:\n%s", exc)
+
+    def run_benchmarks(self):
+        try:
+            parallel = ParallelObject(self._benchmark_runners, timeout=300)
+            parallel.run(lambda x: x.run_benchmarks(), ignore_exceptions=True)
+        except TimeoutError as exc:
+            LOGGER.warning("Run into TimeoutError during running benchmarks. Exception:\n%s", exc)
+        self._collect_benchmark_output()
+
+    def _collect_benchmark_output(self):
+        """
+        Collect the results from ScyllaClusterBenchmarkRunner
+        instances and post them to Elasticsearch.
+        """
+        test_id = TestConfig().test_id()
+
+        for runner in self._benchmark_runners:
+            if runner.benchmark_results:
+                results = {
+                    "test_id": test_id,
+                    "node_instance_type": runner.node_instance_type,
+                    "node_name": runner.node_name,
+                    **runner.benchmark_results
+                }
+                doc_id = f"{test_id}-{runner.node_name.split('-')[-1]}"
+                self._es.create_doc(index=ES_INDEX, doc_type=None, doc_id=doc_id, body=results)
+            else:
+                LOGGER.info("No benchmarks results for node: %s", runner.node_name)
+
+
+class ScyllaNodeBenchmarkRunner:
+    # pylint: disable=broad-except
+    """
+    ScyllaNodeBenchmarkRunner installs and runs benchmarking
+    tools on given cluster nodes and collects the output.
+    """
+
+    def __init__(self, node: "BaseNode"):
+        self._node = node
+        self._remoter: RemoteCmdRunnerBase = node.remoter
+        self.node_instance_type = self._get_db_node_instance_type()
+        self._benchmark_results = {}
+
+    @property
+    def node_name(self):
+        return self._node.name
+
+    @property
+    def benchmark_results(self):
+        return self._benchmark_results
+
+    def install_benchmark_tools(self):
+        clone_repo(self._remoter, "https://github.com/akopytov/sysbench.git")
+        clone_repo(self._remoter, "https://github.com/KnifeyMoloko/cassandra-fio.git")
```

Comment:
please add a comment here, why we are using a private fork. (and a link to the PR to upstream, if any)

Reply:
@fruch I moved to using my own fork here sine the upstream repo was not updated since 2017 and it had a few small issues I wanted to fix for own needs (e.g. making report format default to JSON). I've added a comment with the upstream repo's URL.